Utility/DynamicJsonParser.cs

```diff
@@ -135,6 +137,73 @@ private static dynamic GetDynamic(PSObject obj, string numericalPrefix = "n", bo
                     propertyName = propertyName.ToLowerInvariant();
                 }
 
+                // Nuspec contains xml metadata as a string (only applicable to VSTS feeds)
+                // Example nuspec property:
+                // "nuspec":"<?xml version=\"1.0\" encoding=\"utf-8\"?>
+                //           <package xmlns=\"http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd\">
+                //                <metadata>
+                //                      <id>Microsoft.Kusto.Tools</id>
+                //                      <version>1.0.6</version>
+                //                      <title>Kusto Tools</title>
+                //                      <authors>Microsoft</authors>
+                //                      <owners>Microsoft</owners>
+                //                      <requireLicenseAcceptance>false</requireLicenseAcceptance>
+                //                      <licenseUrl>http://kusto.blob.core.windows.net/kusto-nuget/EULA-agreement.htm</licenseUrl>
+                //                      <projectUrl>http://aka.ms/kdocs</projectUrl>
+                //                      <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288890</iconUrl>
+                //                      <description>Kusto Tools</description>
+                //                      <releaseNotes>1.0.6: LightIngest refactored</releaseNotes>
+                //                      <copyright>Copyright ©  Microsoft Corporation</copyright>
+                //              </metadata>
+                //          </package>"
+                if (string.Equals(propertyName, "nuspec", StringComparison.OrdinalIgnoreCase))
+                {
+                    string xmlString = (string)psProperty.Value;
+
+                    if (string.IsNullOrWhiteSpace(xmlString))
+                    {
+                         var message = string.Format(Messages.InvalidNuspec, "xmlString");
+                         throw new InvalidDataException(message);
+                    }
+
+                    // Remove any characters that may come before the xml tag
+                    // Characters may appear before the start of an xml tag if xml encoding is specified
+                    if ((xmlString[0] != '<') && xmlString.Contains('<'))
```

Comment:
@alerickson I don't think this logic works. You may consider something like following.
``` C#
if (string.Equals(propertyName, "nuspec", StringComparison.OrdinalIgnoreCase))
{
	
	string xmlString = psProperty.Value as string
	
	if (!string.IsNullOrWhiteSpace(xmlString))
	{

 		if ((xmlString[0] != '<') && xmlString.Contains('<'))
                {
                     var strIndex = xmlString.IndexOf('<');
                     xmlString = xmlString.Substring(strIndex, xmlString.Length - strIndex);
               }

       }

      	 if (string.IsNullOrWhiteSpace(xmlString))
         {
                var message = string.Format(Messages.InvalidNuspec, "xmlString");
                throw new InvalidDataException(message);
         }


....


}
```

Reply:
I did change it from what we initially discussed because I realized a string such as "????" 
would throw an index out of bounds error which checking for the index of '<'.  That means there shouldn't be a situation in which a string would hit the second "if (string.IsNullOrWhiteSpace(xmlString))" condition.